wrappers/wrapper.rb

```diff
@@ -169,13 +144,22 @@ def assert_correctness_matrices_vehicles_and_points_definition(vrp)
     def assert_one_sticky_at_most(vrp)
       (vrp.services.empty? || vrp.services.none?{ |service|
         service.sticky_vehicles.size > 1
-      }) && (vrp.shipments.empty? || vrp.shipments.none?{ |shipment|
-        shipment.sticky_vehicles.size > 1
       })
     end
 
-    def assert_no_relations(vrp)
-      vrp.relations.empty? || vrp.relations.all?{ |relation| relation.linked_ids.empty? && relation.linked_vehicle_ids.empty? }
+    def assert_no_relations_except_simple_shipments(vrp)
+      vrp.relations.all?{ |r|
+        return true if r.linked_ids.empty? && r.linked_vehicle_ids.empty?
+        return false unless r.type == :shipment && r.linked_ids.size == 2
+
+        quantities = Hash.new {}
+        vrp.units.each{ |unit| quantities[unit.id] = [] }
+        r.linked_services.first.quantities.each{ |q| quantities[q.unit.id] << q.value }
+        r.linked_services.last.quantities.each{ |q| quantities[q.unit.id] << q.value }
+        quantities.all?{ |_unit, values|
+          [0, 2].include?(values.size) && values.first >= 0 && values.first == -values.last
```

Comment:
This check should be done in vrp.create

Reply:
OR-Tools can handle a shipment with a pickup of -1 and a delivery of -2. There no reason to refuse a problem with such quantities